plugins/vpc-bridge/config/k8s/connector_binary.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector && windows
+
+package k8s
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+
+	log "github.com/cihub/seelog"
+)
+
+const (
+	// Represents Env variable for k8s connector binary path.
+	envK8sConnectorBinaryPath = "AWS_VPC_CNI_K8S_CONNECTOR_BINARY_PATH"
+	// Represents default path for k8s connector binary in EKS Windows AMIs.
+	defaultK8sConnectorBinaryPath = `C:\Program Files\Amazon\EKS\cni\aws-vpc-cni-k8s-connector.exe`
+
+	// Represents Env for log level for k8s connector binary.
+	envLogLevel = "VPC_CNI_LOG_LEVEL"
+	// Represents default log level for k8s connector binary.
+	defaultLogLevel = "info"
+)
+
+// ConnectorBinary Represents k8s connector binary.
+type ConnectorBinary struct {
+	BinaryPath, BinaryLogLevel string
+}
+
+// newConnectorBinary returns connector binary.
+func newConnectorBinary() ConnectorBinary {
+	return ConnectorBinary{
+		getK8sConnectorBinaryPath(), // Get Path to aws-vpc-cni-k8s-connector binary from env variable.
+		getK8sConnectorLogLevel(),   // Use VPC CNI log level for k8s connector binary.
+	}
+}
+
+// executeK8sConnector executes aws-vpc-cni-k8s-connector binary to get pod IP.
+// Execution logs from binary are in stdout. Output from binary is received over named pipe passed as arg.
+func (c *ConnectorBinary) executeK8sConnector(p Pod, pipe string) error {
+
+	// Prepare command to execute binary with required args.
+	cmd := exec.Command(c.BinaryPath,
+		"-pod-name", p.PodName, "-pod-namespace", p.PodNamespace,
+		"-pipe", pipe, "-log-level", c.BinaryLogLevel)
+	log.Debugf("Executing cmd: %s to get pod IP", cmd.String())
+
+	var errBytes, outputBytes bytes.Buffer
+	cmd.Stderr = &errBytes
+	cmd.Stdout = &outputBytes
+
+	// Run command
+	err := cmd.Run()
+
+	log.Infof("logs from k8s connector binary...\n")
+	log.Infof("%s\n", outputBytes.String())
+	log.Infof("end of k8s connector binary logs\n")
+
+	if err != nil || cmd.ProcessState.ExitCode() != 0 {
+		return fmt.Errorf("error executing connector binary: %w with execution error: %s", err, errBytes.String())
```

Comment:
These are the same check. But can we log the exit code alongside error and out

Reply:
I will log the exit code too along side the error.

Reply:
exit code is already part of err. No need to log it.